txcache/transactionsTracker.go

```diff
@@ -0,0 +1,127 @@
+package txcache
+
+import (
+	"math"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+)
+
+// accountRange defines a structure which contains the minimum and the maximum nonce of an account
+type accountRange struct {
+	minNonce core.OptionalUint64
+	maxNonce core.OptionalUint64
+}
+
+// transactionsTracker contains a map of accounts and for each account corresponds a (minNonce, maxNonce) range
+type transactionsTracker struct {
+	accountsWithRange map[string]*accountRange
+}
+
+// newTransactionsTracker creates a new transactions tracker
+func newTransactionsTracker(tracker *selectionTracker, transactions []*WrappedTransaction) *transactionsTracker {
+	txTracker := &transactionsTracker{
+		accountsWithRange: make(map[string]*accountRange),
+	}
+	txTracker.createAccountsWithDefaultRange(transactions)
+	txTracker.updateAccountsWithRange(tracker)
+
+	return txTracker
+}
+
+// createAccountsWithDefaultRange initializes all the required accounts with a default range
+func (txTracker *transactionsTracker) createAccountsWithDefaultRange(transactions []*WrappedTransaction) {
+	for _, tx := range transactions {
+		if tx == nil || tx.Tx == nil {
+			continue
+		}
+
+		sender := tx.Tx.GetSndAddr()
+		_, ok := txTracker.accountsWithRange[string(sender)]
+		if !ok {
+			txTracker.accountsWithRange[string(sender)] = &accountRange{
+				minNonce: core.OptionalUint64{Value: math.MaxUint64, HasValue: false},
+				maxNonce: core.OptionalUint64{Value: 0, HasValue: false},
+			}
+		}
+	}
+}
+
+// updateAccountsWithRange updates all the saved accounts with the range extracted from breadcrumbs
+func (txTracker *transactionsTracker) updateAccountsWithRange(tracker *selectionTracker) {
+	tracker.mutTracker.RLock()
```

Comment:
I see that instances of `transactionsTracker` have a short lifespan, they are ephemeral (variables that refer them only live on the stack). Is a mutex necessary / is there any concurrent access?

Reply:
You can create a simple getter for the blocks, on the `selectionTracker`. And that getter would do a mere slice clone.

Can be done here, on in the next PR, but should be done.